code/contracts/ONEWallet2.sol

```diff
@@ -0,0 +1,176 @@
+
+    /**
+     * It serves for the deletion of accidentaly submitted commits by the user or malicious commits submitted by the attackers, which contribute to the total storage allocation.
+     * Currently, this method can be called by anybody since it does touch only the garbage. 
+     * Also, in theory, someone may earn gas if there are too many entries submitted by the attacker (who payed for it).
+     */
+    function cleanUpCommits() external {                 
+        for (uint32 i = lastCleanUpOfCommitsIdx; i <= lastRevealedOTPIdx; i++){ // IH: in theory, this might lead to out-of-gas exception for settings of TOTP with very long validity but currently I'd not be affraid. Moreover, we are  getting the gas back for successfull deletions.
+            delete commits[i]; // NOTE: I'd say it makes no difference whether it was initialized or not, so we can delete any entry.        
+        }
+        lastCleanUpOfCommitsIdx = lastRevealedOTPIdx; // shift commit clean-up checkpoint
+    }
+
+    function getCurrentSpending() external view returns (uint){ 
+        return dailySpendings[block.timestamp / 1 days];
+    }        
+
+    function retire() external returns (bool){
+        require(uint32(block.timestamp / interval) - t0 > lifespan, "Too early to retire");        
+        return _drain();
+    }
+
+    //////////////////////// Internal Functions ////////////////////////// 
+
+    /**
+     * It cleans up the previous value of daily spending from the map (this ensures that map has always only a single value -- except the execution time of the reveal())
+     * Can be called only after successfully executed transfer within reveal().
+     */      
+    function _cleanUpDailySpendings() internal {
+        uint32 curDay = uint32(block.timestamp / 1 days);
+        if(curDay > lastExecutedTransfer ){
+            delete dailySpendings[lastExecutedTransfer];
+        }
+        lastExecutedTransfer = curDay; // shift the clean-up checkpoint
+    }    
+
+    function _verifyMerkleProof(bytes32[] calldata neighbors, uint32 index, bytes32 eotp) internal view
+    {                
+        require(neighbors.length == height - 1, "Not enough neighbors provided");
+        bytes32 h = keccak256(abi.encodePacked(eotp));
+        for (uint8 i = 0; i < height - 1; i++) {
+            if ((index & 0x01) == 0x01) {
+                h = keccak256(abi.encodePacked(neighbors[i], h));                
+            } else {
+                h = keccak256(abi.encodePacked(h, neighbors[i]));
+            }
+            index >>= 1;
+        }
+        require(root == h, "Proof is incorrect");        
+    }
+
+    function _executeTransfer(address payable dest, uint amount) internal returns(bool) {
+        if(dailySpendings[block.timestamp / 1 days] + amount > dailyLimit)
+            return false; // Daily limit exceeded.        
+        
+        if(!dest.send(amount))
```

Comment:
`send` only forward 2300 gas, which is insufficient if `dest` is a contract and implements non-trivial logics in `receive()`. I changed it to `.call{value: amount}()` in my implementation in an early version.

Reply:
I see. Fixed.